### DISCUSSION TITLE
Improve AI tag management

### DISCUSSION THREAD

**Discussion by ant1fr** [karakeep-app/karakeep#350] [Ideas]
Hoarder is an excellent app, and I'm particularly impressed with the AI Tags generated by ChatGPT. However, I've noticed that ChatGPT sometimes produces slightly different tags for the same concept (e.g., "open-source app," "open-source," "open source").

To address these near-duplicate tags, I suggest a few potential solutions:

1. Prompt Adjustment: Modify the prompt to include existing tags and instruct ChatGPT to select from them when appropriate. Pros: This approach keeps tagging fully automated and more consistent. Cons: It could increase token usage over time.
2. User-Validated Tags: Enhance the tag management system to allow users to "validate" AI-generated tags by dragging them from "AI Tags" to "My Tags." These user-approved tags could then be used in future prompts. Pros: Less token-intensive than option 1. Cons: Requires user involvement, which may be time-consuming.

Additionally, a complementary approach could involve periodic tag review and standardization. This would entail running a specific prompt that provides ChatGPT with all the AI-generated tags, asking it to suggest merges, clean up, and standardize the tags.

**Comment by huyz** [comment]
See https://github.com/hoarder-app/hoarder/issues/111

**Reply by ant1fr** [reply]
Thanks! Missed that one!
